exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -1682,17 +1746,54 @@ func (b *Bitfinex) Unsubscribe(channelsToUnsubscribe []stream.ChannelSubscriptio
 			"chanId": chanID,
 		}
 
-		err := b.Websocket.Conn.SendJSONMessage(req)
+		respRaw, err := b.Websocket.Conn.SendMessageReturnResponse("unsubscribe:"+strconv.Itoa(chanID), req)
 		if err != nil {
 			errs = common.AppendError(errs, err)
 			continue
 		}
-		// We do this before the unsubscribed event comes back so we can subscribe again when called from ResubcribeToChannel
+
+		if err := b.getErrResp(respRaw); err != nil {
+			wErr := fmt.Errorf("error unsubscribing from ChanId: %v; %w", chanID, err)
+			b.Websocket.DataHandler <- wErr
+			return wErr
+		}
+
+		b.wsSubMutex.Lock()
+		delete(b.wsSubChannels, chanID)
+		b.wsSubMutex.Unlock()
 		b.Websocket.RemoveSuccessfulUnsubscriptions(channelsToUnsubscribe[i])
 	}
 	return errs
 }
 
+// getErrResp takes a json response string and looks for an error event type
+// If found it parses the error code and message as a wrapped error and returns it
+// It might log parsing errors about the nature of the error
+// If the error message is not defined it will return a wrapped errUnknownError
+func (b *Bitfinex) getErrResp(resp []byte) error {
+	event, err := jsonparser.GetUnsafeString(resp, "event")
+	if err != nil {
+		return fmt.Errorf("error parsing WS event: %w from message: %s", err, resp)
+	}
+
+	if event != "error" {
+		return nil
+	}
+	errCode, err := jsonparser.GetInt(resp, "code")
+	if err != nil {
+		log.Errorf(log.ExchangeSys, "%s error parsing WS error code: %s from message: %s", b.Name, err, resp)
```

Comment:
We log this instead of returning it because we already have an error, and this shouldn't distract from that.
Also doesn't seem likely.